File: client-sdks/sockudo-kotlin/lib/src/main/kotlin/io/sockudo/client/ProtocolPrefix.kt
package io.sockudo.client

/**
 * Protocol version prefix helpers.
 *
 * Protocol v1 uses the original Pusher event prefix ("pusher:" / "pusher_internal:").
 * Protocol v2 (Sockudo-native, the default) uses "sockudo:" / "sockudo_internal:".
 */
class ProtocolPrefix(val version: Int) {
    val eventPrefix: String = if (version >= 2) "sockudo:" else "pusher:"
    val internalPrefix: String = if (version >= 2) "sockudo_internal:" else "pusher_internal:"

    fun event(name: String): String = "$eventPrefix$name"
    fun internal_(name: String): String = "$internalPrefix$name"

    fun isInternalEvent(name: String): Boolean = name.startsWith(internalPrefix)
    fun isPlatformEvent(name: String): Boolean = name.startsWith(eventPrefix)
}

File: client-sdks/sockudo-flutter/lib/src/protocol_prefix.dart
/// Protocol version prefix helpers.
///
/// Protocol v1 uses the original Pusher event prefix ("pusher:" / "pusher_internal:").
/// Protocol v2 (Sockudo-native) uses "sockudo:" / "sockudo_internal:".
class ProtocolPrefix {
  ProtocolPrefix(int version) : _version = version {
    if (version == 2) {
      _eventPrefix = 'sockudo:';
      _internalPrefix = 'sockudo_internal:';
    } else {
      _eventPrefix = 'pusher:';
      _internalPrefix = 'pusher_internal:';
    }
  }

  final int _version;
  late final String _eventPrefix;
  late final String _internalPrefix;

  int get version => _version;
  String get eventPrefix => _eventPrefix;
  String get internalPrefix => _internalPrefix;

  String event(String name) => '$_eventPrefix$name';
  String internal(String name) => '$_internalPrefix$name';

  bool isInternalEvent(String name) => name.startsWith(_internalPrefix);
  bool isPlatformEvent(String name) => name.startsWith(_eventPrefix);
}
